ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -40,3 +56,292 @@ func TestFaultLatencyFaultPath(t *testing.T) {
 func TestFaultPacketLossFaultPath(t *testing.T) {
 	assert.Equal(t, "/api/{endpointContainerIDMuxName:[^/]*}/fault/v1/network-packet-loss", FaultNetworkFaultPath(types.PacketLossFaultType))
 }
+
+type networkBlackHolePortTestCase struct {
+	name                      string
+	expectedStatusCode        int
+	requestBody               interface{}
+	expectedResponseBody      types.NetworkFaultInjectionResponse
+	setAgentStateExpectations func(agentState *mock_state.MockAgentState)
+}
+
+func getNetworkBlackHolePortTestCases(name string, expectedHappyResponseBody types.NetworkFaultInjectionResponse) []networkBlackHolePortTestCase {
+	happyBlackHolePortReqBody := map[string]interface{}{
+		"Port":        port,
+		"Protocol":    protocol,
+		"TrafficType": trafficType,
+	}
+	tcs := []networkBlackHolePortTestCase{
+		{
+			name:                 fmt.Sprintf("%s success", name),
+			expectedStatusCode:   200,
+			requestBody:          happyBlackHolePortReqBody,
+			expectedResponseBody: expectedHappyResponseBody,
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil)
+			},
+		},
+		{
+			name:               fmt.Sprintf("%s unknown request body", name),
+			expectedStatusCode: 400,
+			requestBody: map[string]interface{}{
+				"Port":        port,
+				"Protocol":    protocol,
+				"TrafficType": trafficType,
+				"Unknown":     "",
```

Comment:
Can we allow unknown fields? and then make sure only required fields are present?